pageserver/src/bin/pageserver.rs

```diff
@@ -284,9 +284,10 @@ fn start_pageserver(conf: &'static PageServerConf, daemonize: bool) -> Result<()
         None,
         "http_endpoint_thread",
         true,
-        move || {
+        move || -> Result<(), error::Error> {
```

Comment:
It's a bit odd that an error enum with "postgres" and "internal" variants is used in an **http** thread, we clearly do regular http without postgres involved, so there's something odd here.

Reply:
Cannot agree more. This is a result of the lack of typed errors and an attempt to add one without many error variants.

Right now, all the errors returned by this http thread will be categorized as internal.